### PR TITLE
Feat: NJT Rail Translator and interface parsing changes

### PR DESCRIPTION
**Pull request by doellerbe06**
This translator accepts data from NJT's proprietary Train Control system. It produces trip updates similar to GTFS realtime format but with some additional context in the absence of some required and conditionally required GTFS-RT fields.

Translators will also expect raw data that they will parse in the format they expect.
- MTA, SEPTA, LA => JSON
- NJT => XML

Minor version bump to v0.5.0

### CODE REVIEW

File: gtfs_realtime_translators/bindings/intersection_pb2.py [hunk 1]
```diff
@@ -134,18 +175,18 @@
 DESCRIPTOR.extensions_by_name['intersection_stop_time_update'] = intersection_stop_time_update
 _sym_db.RegisterFileDescriptor(DESCRIPTOR)
 
-IntersectionTripUpdate = _reflection.GeneratedProtocolMessageType('IntersectionTripUpdate', (_message.Message,), {
-  'DESCRIPTOR' : _INTERSECTIONTRIPUPDATE,
-  '__module__' : 'intersection_pb2'
+IntersectionTripUpdate = _reflection.GeneratedProtocolMessageType('IntersectionTripUpdate', (_message.Message,), dict(
+  DESCRIPTOR = _INTERSECTIONTRIPUPDATE,
```
Comment (devadvance) [inline gtfs_realtime_translators/bindings/intersection_pb2.py]: Just to make sure: this was done by the proto -> py conversion automatically, not by hand, right?

File: gtfs_realtime_translators/translators/njt_rail.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            'northeast_corridor_line': '9',
+            'north_jersey_coast_line': None,
+        }
+
+        key = 'amtrak' if data['line_abbreviation'] == 'AMTK' else data['line'].replace(' ', '_').lower()
+        route_id = route_id_lookup.get(key, None)
+        if route_id is not None:
+            return route_id
+
+        def get_route_id_by_origin_or_destination(line_key, line_name, origin, destination):
```
Comment (devadvance) [inline gtfs_realtime_translators/translators/njt_rail.py]: This looks fine. We should note these in our internal documentation for product/design to be aware.

File: gtfs_realtime_translators/bindings/intersection_pb2.py (hunk 1 above)
Comment (doellerbe06) [reply gtfs_realtime_translators/bindings/intersection_pb2.py]: I didn't manually change these files, but I'll double-check that the IDE didn't modify them.

File: gtfs_realtime_translators/translators/njt_rail.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+class NjtRailGtfsRealtimeTranslator:
+    ROUTE_ID_LOOKUP = {
+        'atlantic_city_ine': '1',
+        'montclair-boonton_line': None,
+        'main_line': '5',
+        'bergen_county_line': '6',
+        'morristown_line': '7',
+        'gladstone_branch': '8',
+        'northeast_corridor_line': '9',
+        'north_jersey_coast_line': None,
```
Comment (joerodrig) [inline gtfs_realtime_translators/translators/njt_rail.py]: Just clarifying: Does none just mean realtime data doesn't exist for this particular route, even though the route exists in the static GTFS?

Comment (joerodrig) [reply gtfs_realtime_translators/translators/njt_rail.py]: I think I see what's going on. This value can be set to None if (1) There is no Realtime GTFS feed for the route or (2) There is a GTFS feed, but the routeID can change based on destination, so we're calculating that on [L#97](https://github.com/Intersection/gtfs-realtime-translators/pull/6/commits/ce6a6224fb0cc3cfd4c653107d20cf55e7f75c8f#diff-60762aaa1cfc79422090c3046168973fR97)

Comment (doellerbe06) [reply gtfs_realtime_translators/translators/njt_rail.py]: Good question. And yes. Just to clarify, the routes indeed exist in the GTFS data, but we have to extrapolate this ID in certain cases based on their realtime data and what is truth in their GTFS data.

File: gtfs_realtime_translators/bindings/intersection_pb2.py (hunk 1 above)
Comment (doellerbe06) [reply gtfs_realtime_translators/bindings/intersection_pb2.py]: Confirmed this was only modified from the conversion.

**Pull request merged**